src/index.ts

```
@@ -415,6 +432,87 @@ export class FlashbotsBundleProvider extends providers.JsonRpcProvider {
+
+    //Check if carrier_tx has minimum params, populate with defaults if not
+    /*
+     The following statement is intended to be used in order to support any type of incomplete TransactionRequest
+     received, populating it with default values if any one is missing
+     */
+    await this.prepareCarrierTransaction(carrierTx, signer)
+
+    //Populate carrier_tx.data as : carrier_tx.data = MEV_Prefix | validator pub_key | Encrypt(validator pub_key, serialized bundle)
+    const mevPrefix = `0123` //this is a placeholder!
+
+    let payload = `0x`
+    payload += mevPrefix
+    payload += validatorPublicKey
+    payload += encryptedBundle.toString('hex')
+
+    carrierTx.data = payload
+
+    //Sign the transaction received as param with passed signer
+    const signedTx = await signer.signTransaction(carrierTx)
+
+    //Propagate carrier_tx into the public mempool and return Promise<TransactionResponse> for the carrier_tx
+    return this.genericProvider.sendTransaction(signedTx)
+  }
+
+  /**
+   * A private method to encode a FlashbotsBundle following the RLP serialization standard
+   * @param bundle the FlashbotsBundle instance to be serialized
+   * @return string the rlp encoded bundle
+   * @private
+   */
+  private rlpSerializeBundle(bundle: FlashbotsBundle): string {
+    const stringifiedBundle = JSON.stringify(bundle)
+    const bytesBundle = Buffer.from(stringifiedBundle)
+    return encode(bytesBundle)
+  }
+
+  /**
+   * A private method to populate {@param carrier}'s missing fields with default values
+   * @param carrier an instance of TransactionRequest which will be the tx containing the full payload in its data field
+   * @param signer an instance of Signer, which is used to retrieve the correct nonce for the carrier tx, should it be missing
+   * @private
+   */
+  private async prepareCarrierTransaction(carrier: TransactionRequest, signer: Signer) {
+    if (!('to' in carrier)) carrier.to = '0x0000000000000000000000000000000000000000'
```

we should probably throw an error if they didn't fill in a "to" field

---

acknoweledged, will refactor this